### ISSUE TITLE
Parameter --ami-id is not working properly

### ISSUE THREAD

**Issue by jweyrich** [kubernetes-incubator/kube-aws#764]
Hi,

I'm generating a `cluster.yaml` with the following command:

`kube-aws init --region=us-east-1 --cluster-name=test-cluster --availability-zone=us-east-1a --external-dns-name=cluster.example.com --key-name="cluster-key" --kms-key-arn="arn:aws:kms:us-east-1:073xxx..." --ami-id="ami-xxxx"`

However, the generated file doesn't reference the provided AMI anywhere. It outputs as below:

```
# The AMI ID of CoreOS.
# If omitted, the latest AMI for the releaseChannel is used.
#amiId: ""
```

kube-aws version 0.9.7

Can anyone confirm? For now I'm changing it manually.

**Comment by redbaron** [comment]
Not to say that it shouldn't be fixed, but wouldn't using coreos release channels work  for you instead?

**Comment by jweyrich** [comment]
I believe it wouldn't. There's a requirement for encrypted volumes (using KMS). AFAIK the only way to achieve that would be using a pre-encrypted AMI - or there's another way? Thank you for the suggestion though.

**Comment by redbaron** [comment]
kube-aws supports creating and attaching extra data volumes, I think encryption support for such EBS volumes can be added relatively easy. You then have unencrypted root using upstream CoreOS AMIs and encrypted data volumes, will it be enough to meet your requirements?

**Comment by jweyrich** [comment]
Unfortunately not. They require the root volume to be encrypted as well.
I found important to report because others might blindly trust the CLI parameter.

**Comment by mumoshu** [comment]
@jweyrich Hi, thanks for trying kube-aws!
Yes, it does seem like a bug. I'll fix it until v0.9.8-rc.1.

Btw, you will probably want to add `--hosted-zone-id` too so that you no longer need to add `hostedZoneId` to your cluster.yaml afterwards.

**Comment by jweyrich** [comment]
Hmm, I'll check the parameter you mentioned. Thank you for the fix! ;-)
